### PR TITLE
build: Generate automatic documentation

### LINKED ISSUE #74: Generate automated documentation using GitHub Actions
Generate automated documentation using GitHub Actions

### PR DESCRIPTION
**Pull request by aadarsh-ram**
**Related issue**
- Fixes #74 

**Solution**
- Uses mkdocs for static website generation
- Creates a new branch gh-pages automatically after workflow run

If the PR is merged, #85 can be closed. Hope to contribute more!

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Open Food Facts client for your Python applications and scripts
```
Comment (alexgarel) [inline docs/index.md]: This should be a symlink to ../README.md so that updates are automatic.

Comment (aadarsh-ram) [reply docs/index.md]: I don't think symlinks get rendered in the website. I instead copy README.md each time, before mkdocs website generation. This way updates are automatic.

File: .github/workflows/generate-docs.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  # Triggers the workflow on push
+  push:
+    branches:
+      - actions-dev
+      - develop
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
```
Comment (alexgarel) [inline .github/workflows/generate-docs.yml]: you can remove this kind of comments that explains how workflow works. Just keep comments that make sens.

File: .github/workflows/generate-docs.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v2
+        with: 
+          python-version: 3.8
+      - run: cp ./README.md ./docs/index.md
```
Comment (alexgarel) [inline .github/workflows/generate-docs.yml]: Good idea :+1:

**Review by alexgarel — APPROVED** [review]
Ok, I'm not really able to control (don't know about mkdocs) but I trust you :-)

If you can remove comments that are not really useful.

File: .github/workflows/generate-docs.yml (hunk 2 above)
Comment (aadarsh-ram) [reply .github/workflows/generate-docs.yml]: Yep, I'll do this ASAP.

Comment (aadarsh-ram) [reply .github/workflows/generate-docs.yml]: Yep, I added a name to each step so that it's easier to understand.

**Comment by teolemon** [comment]
That's great, I've enabled GitHub Pages, and we can merge

**Comment by aadarsh-ram** [comment]
@teolemon After the "Generate automatic documentation" workflow completes in the main repo (after merge), you might need to configure github-pages with "gh-pages" as the source branch. Root ("/") should be the folder.

**Comment by teolemon** [comment]
https://openfoodfacts.github.io/openfoodfacts-python/ @aadarsh-ram

**Comment by aadarsh-ram** [comment]
@teolemon Please do make sure the branch is "gh-pages" and folder "/" in the settings of github-pages.

**Comment by teolemon** [comment]
Fixed 👍

**Comment by alexgarel** [comment]
kudos @aadarsh-ram that's really cool :-)

**Pull request merged**